File: bank-account/python/tests/fixed_clock.py
from __future__ import annotations

from datetime import date


class FixedClock:
    def __init__(self, current: date) -> None:
        self._current = current

    def today(self) -> date:
        return self._current

    def advance_to(self, d: date) -> None:
        self._current = d

File: bank-account/csharp/tests/BankAccount.Tests/FixedClock.cs
namespace BankAccount.Tests;

public class FixedClock : IClock
{
    private DateOnly _today;

    public FixedClock(DateOnly today) => _today = today;

    public DateOnly Today() => _today;

    public void AdvanceTo(DateOnly date) => _today = date;
}
